crates/amalthea/src/kernel.rs

```diff
@@ -33,42 +35,17 @@ use crate::socket::socket::Socket;
 use crate::socket::stdin::StdInRequest;
 use crate::socket::stdin::Stdin;
 use crate::stream_capture::StreamCapture;
+use crate::wire::handshake_request::HandshakeRequest;
 use crate::wire::input_reply::InputReply;
+use crate::wire::jupyter_message::JupyterMessage;
 use crate::wire::jupyter_message::Message;
 use crate::wire::jupyter_message::OutboundMessage;
+use crate::wire::jupyter_message::Status;
 
 macro_rules! report_error {
     ($($arg:tt)+) => (if cfg!(debug_assertions) { panic!($($arg)+) } else { log::error!($($arg)+) })
 }
 
-/// A Kernel represents a unique Jupyter kernel session and is the host for all
-/// execution and messaging threads.
-pub struct Kernel {
-    /// The name of the kernel.
-    name: String,
-
-    /// The connection metadata.
-    connection: ConnectionFile,
-
-    /// The unique session information for this kernel session.
-    session: Session,
-
-    /// Sends messages to the IOPub socket. This field is used throughout the
-    /// kernel codebase to send events to the frontend; use `create_iopub_tx`
-    /// to access it.
-    iopub_tx: Sender<IOPubMessage>,
-
-    /// Receives message sent to the IOPub socket
-    iopub_rx: Option<Receiver<IOPubMessage>>,
-
-    /// Sends notifications about comm changes and events to the comm manager.
-    /// Use `create_comm_manager_tx` to access it.
-    comm_manager_tx: Sender<CommManagerEvent>,
-
-    /// Receives notifications about comm changes and events
-    comm_manager_rx: Receiver<CommManagerEvent>,
-}
```

Comment:
We were able to remove all of this state from Amalthea's `Kernel`. For instance the IOPub and CommManager channels are now created by the caller of`connect()`.

That's why the diff in this file is pretty big. We removed the `Kernel` struct altogether and all impls are now free functions in the `kernel` module. Most of the diff comes from the ensuing unnesting.